### PR TITLE
Add OpenSLR dataset

### PR DESCRIPTION
**Pull request by cahya-wirawan**
OpenSLR (https://openslr.org/) is a site devoted to hosting speech and language resources, such as training corpora for speech recognition, and software related to speech recognition. There are around 80 speech datasets listed in OpenSLR, currently this PR includes only 9 speech datasets SLR41, SLR42, SLR43, SLR44, SLR63, SLR64, SLR65, SLR66 and SLR69 (Javanese, Khmer, Nepali and Sundanese, Malayalam, Marathi, Tamil, Telugu and Catalan). I can add other speech datasets gradually next time.

### CODE REVIEW

File: datasets/openslr/openslr.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                    # Following regexs are needed to normalise the lines, since the datasets
+                    # are not always consistent and have bugs:
+                    line = re.sub(r"\t[^\t]*\t", "\t", line.strip())
+                    field_values = re.split(r"\t\t?", line)
+                    # set absolute path for audio file
+                    if len(field_values) != 2:
+                        continue
+                    field_values[0] = os.path.join(path_to_datas[i], f"{field_values[0]}.wav")
+                    counter += 1
+                    yield counter, {key: value for key, value in zip(data_fields, field_values)}
```
Comment (lhoestq) [inline datasets/openslr/openslr.py:216]: Maybe separating the path from the sentence would make this code easier to read ?

```suggestion
                    field_values = re.split(r"\t\t?", line)
                    if len(field_values) != 2:
                        continue
                    filename, sentence = field_values
                    # set absolute path for audio file
                    path = os.path.join(path_to_datas[i], f"{filename}.wav")
                    counter += 1
                    yield counter, {"path": path, "sentence": sentence}
```

File: datasets/openslr/README.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+## Dataset Structure
+
+### Data Instances
+
+A typical data point comprises the path to the audio file, called path and its sentence. 
+
+#### SLR41, SLR42, SLR43, SLR44, SLR63, SLR64, SLR65, SLR66, SLR69  
+```
+{
+  'path': 'suf_00297_00037352660'
```
Comment (lhoestq) [inline datasets/openslr/README.md:201]: isn't this supposed to be a path to a wav file ?

**Review by lhoestq — COMMENTED** [review]
This is really cool thank you !

File: datasets/openslr/README.md (hunk 2 above)
Comment (cahya-wirawan) [reply datasets/openslr/README.md:201]: yes, this should be a path to a wav file in .cache directory. I will update the documentation accordingly, thanks.

File: datasets/openslr/README.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+multilinguality:
+- multilingual
+size_categories:
+- 100K<n<1M
+source_datasets:
+- openslr
+task_categories:
+- other
+task_ids:
+- other-other-automatic-speech-recognition
```
[the remark this answers is not in the document]
Comment (cahya-wirawan) [reply datasets/openslr/README.md:27]: According to the LICENSE files in each datasets listed above (https://www.openslr.org/resources/<SLR number>/LICENSE), they have the same license which is "Creative Commons Attribution-ShareAlike 4.0". However, Google listed also those datasets in their repository https://github.com/google/language-resources#license. They wrote that all original files are licensed under an Apache License, Version 2.0, but some datasets are licensed under a Creative Commons Attribution 4.0 International License (CC BY 4.0) where specifically noted.

File: datasets/openslr/README.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+#### SLR69: Crowdsourced high-quality Catalan multi-speaker speech data set
+This data set contains transcribed high-quality audio of Catalan sentences recorded by volunteers. The data set 
+consists of wave files, and a TSV file (line_index.tsv). The file line_index.tsv contains a anonymized FileID and 
+the transcription of audio in the file.
+
+The data set has been manually quality checked, but there might still be errors.
+
+Please report any issues in the following issue tracker on GitHub. https://github.com/googlei18n/language-resources/issues
+
+See LICENSE file for license information.
```
[the remark this answers is not in the document]
Comment (cahya-wirawan) [reply datasets/openslr/README.md:180]: I can mention the license and make a link to the LICENSE file in openslr for each of the datasets.

File: datasets/openslr/openslr.py (hunk 1 above)
Comment (cahya-wirawan) [reply datasets/openslr/openslr.py:216]: Thanks for the suggestion.

File: datasets/openslr/openslr.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+import re
+
+import datasets
+
+
+_DATA_URL = "https://openslr.org/resources/{}"
+
+_CITATION = """\
+
+"""
```
[the remark this answers is not in the document]
Comment (cahya-wirawan) [reply datasets/openslr/openslr.py:29]: I can't really find a citation about OpenSLR, I only found citations for each datasets.

[the remark this answers is not in the document]
Comment (cahya-wirawan) [reply datasets/openslr/openslr.py:29]: should I put all datasets citations there? They are written already in README file btw.

File: datasets/openslr/openslr.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+_DESCRIPTION = """\
+OpenSLR
```
[the remark this answers is not in the document]
Comment (cahya-wirawan) [reply datasets/openslr/openslr.py]: done

File: datasets/openslr/openslr.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets/openslr/openslr.py:29]: Yes please ! Once this is done, we can merge IMO :)

thanks again !

[the remark this answers is not in the document]
Comment (cahya-wirawan) [reply datasets/openslr/openslr.py]: Ok, it's also done

**Review by lhoestq — APPROVED** [review]
LGTM thanks !

I just added a missing section in the readme and updated the language tags

**Pull request merged**